Jenkinsfile new parallel syntax, use artifacts, record test results

davidjsherman opened the pull request:
Jenkinsfile using the new [declarative parallel stages](https://jenkins.io/blog/2017/09/25/declarative-1/) for debian, macos, windows. Relies on davidjsherman/aseba-jenkins@e6c2eb6a6b5e262f072671d04717466945f3fb30 _et seq._. 

Dashel and Enki are not compiled, but retrieved by Jenkins as artifacts from {aseba,enki}-community projects. Rely on CMAKE_FIND_ROOT_PATH to find Dashel and Enki. On macOS there is a hack for Enki [Jenkinsfile#L70](https://github.com/davidjsherman/aseba/blob/master/Jenkinsfile#L70).

Windows dnssd needs ws2_32, see [CMakeLists.txt#L121](https://github.com/davidjsherman/aseba/blob/master/CMakeLists.txt#L121).

Testing results from CTest are recorded, and published using xUnit. Publishing is done in a post stage ([Jenkinsfile#L142](https://github.com/davidjsherman/aseba/blob/master/Jenkinsfile#L142)), so always happens even if the pipeline fails.

stephanemagnenat commented on CMakeLists.txt:
```diff
@@ -116,6 +117,9 @@ if (DNSSD_FOUND)
 	else()
 		set(ASEBA_ZEROCONF_ADDITIONAL_LIBRARIES ${DNSSD_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
+		if (WIN32)
+			set(ASEBA_ZEROCONF_ADDITIONAL_LIBRARIES ${ASEBA_ZEROCONF_ADDITIONAL_LIBRARIES} -lws2_32)
```
I think that this line should go to the CMakeLists.txt specific to Zeroconf that sets `${ASEBA_ZEROCONF_ADDITIONAL_LIBRARIES}`, isn't it?

davidjsherman replied:
OK, I will try that

davidjsherman replied:
In fact, `${ASEBA_ZEROCONF_ADDITIONAL_LIBRARIES}` is set right here in the root CMakeLists.txt, just before the proposed addition. So this seems to be the appropriate place after all.

stephanemagnenat replied:
Ok, right.

davidjsherman commented:
Default parameters don't seem to work for jobs created by GitHub Organization scanning, so the automatic build of PRs always fails. Fixing that now.

stephanemagnenat commented:
@davidjsherman is this PR good to merge?

davidjsherman commented:
Yes. I fixed the failsafe values (15b70ef3324f3f1e6bcffa8adf5a7b5c3a249a08), worked around the `{dashel,enki}_DIR` problem on macOS (dd9b5bbb77fcc00309f79cfeb1e991cd1534b4e2), and added triggers (f536dadcf49e4ea515f87f72b8185f9065dc4cc1) to pick up PRs and changes to Dashel or Enki.

You can see the build and the test results at https://ci.inria.fr/inirobot/job/aseba-community/job/aseba/job/PR-768/lastSuccessfulBuild/.

When this is merged, Jenkins should create jobs for all of the open PRs, at https://ci.inria.fr/inirobot/job/aseba-community/job/aseba/view/change-requests/.

davidjsherman commented:
That file is retrieved from [the Dashel job](https://ci.inria.fr/inirobot/job/aseba-community/job/dashel/job/master/) and has the wrong name there, too.
 
The version wasn't bumped to 1.3.2 in [dashel/debian/changelog](https://github.com/aseba-community/dashel/blob/master/debian/changelog) and as I understand it dpkg-gencontrol uses this to determine the file name. Is the changelog generated, or is it modified by hand?

stephanemagnenat commented:
The debian changelog of Dashel is not updated to 1.3.3 and Dashel is tagged as well.

davidjsherman commented:
Thanks to aseba-community/dashel@576ec8abbac721dfa08be1fd74a7519ffd4fcd5f, libdashel has the correct version in the artifacts compiled for Aseba
